src/main/flight/pid.c

```diff
@@ -719,7 +719,8 @@ void pidController(void)
 {
     uint8_t headingHoldState = getHeadingHoldState();
 
-    if (headingHoldState == HEADING_HOLD_UPDATE_HEADING) {
+   //if (headingHoldState == HEADING_HOLD_UPDATE_HEADING) {
+    if (headingHoldState == HEADING_HOLD_UPDATE_HEADING || headingHoldState == HEADING_HOLD_DISABLED) { //Issue #3177 Heading Hold Yaw Without RC Input
```

Comment:
This is not a proper fix. State `HEADING_HOLD_DISABLE` means that quad is tilted too much and heading control is risky. With proposed change any hard bank or pitch will reset the heading hold target which may lead to drift.

Proper fix would be to change `getHeadingHoldState()` function and modify the logic of detecting `HEADING_HOLD_DISABLED`.

Reply:
Yes, but this fix only applies to recording the yaw current position.  Heading-hold still does not get applied if there is a significant angle.  The problem is when there is a significant angle, we stop updating the current compass position.  If the yaw input is then centered and then the quad levels, heading-hold kicks in and causes the quad to yaw back to the last position before its significant angle.

Reply:
Problem is that if tilt is large you should not apply heading hold but you must keep the desired heading if yaw stick is centered. With your fix any sufficient tilt will reset the heading hold target regardless of yaw stick which is not the desired behavior.

Reply:
Current behavior is to record a new heading hold target is yaw stick is not centered AND tilt is small. This is flawed and needs fixing. 

With your fix we'll record a new heading hold target when yaw stick is not centered OR tilt is large.

Proper logic would be to record a new heading hold target if yaw stick is not centered regardless of tilt.

Reply:
Got it, so we should just update  the get HeadingHoldState:

static uint8_t getHeadingHoldState(void)
{
    // Don't apply heading hold if overall tilt is greater than maximum angle inclination
    if (calculateCosTiltAngle() < headingHoldCosZLimit) {
        if (ABS(rcCommand[YAW]) != 0 && FLIGHT_MODE(HEADING_MODE)) { //Issue #3177 Heading Hold Yaw Without RC Input
            return HEADING_HOLD_UPDATE_HEADING;
        }
        else {
            return HEADING_HOLD_DISABLED;
        }
    }